scipy/stats/_hypotests.py

```diff
@@ -836,52 +836,59 @@ def barnard_exact(table, alternative="two-sided", pooled=True, num_it=3):
     should only be used when both sets of marginals are fixed.
 
     """
-    if num_it <= 0:
+    if n_iter <= 0:
         raise ValueError(
             "Number of iterations `num_it` must be strictly positive, "
-            f"found {num_it!r}"
+            f"found {n_iter!r}"
         )
 
-    c = np.asarray(table, dtype=np.int64)
+    table_nparray = np.asarray(table, dtype=np.int64)
 
-    if not c.shape == (2, 2):
+    if not table_nparray.shape == (2, 2):
         raise ValueError("The input `table` must be of shape (2, 2).")
 
-    if np.any(c < 0):
+    if np.any(table_nparray < 0):
         raise ValueError("All values in `table` must be nonnegative.")
 
-    if 0 in c.sum(axis=0):
+    if 0 in table_nparray.sum(axis=0):
         # If both values in column are zero, the p-value is 1 and
         # the score's statistic is NaN.
         return BarnardExactResult(np.nan, 1.0)
 
-    total_c1, total_c2 = c.sum(axis=0)
+    total_col_1, total_col_2 = table_nparray.sum(axis=0)
 
-    x1 = np.arange(total_c1 + 1, dtype=np.int64).reshape(-1, 1)
-    x2 = np.arange(total_c2 + 1, dtype=np.int64).reshape(1, -1)
+    # We are reshaping x1 and x2 to enable numpy's powerful broadcasting
+    # operators, so that when we are adding or multiplying x1 and x2,
+    # the result is an array of shape (total_col_1 + 1, total_col_2 + 1)
```

Comment:
What I meant here was more to link the math that we read in papers to the implementation. In some simple example, they did not have an array. But just calculated a single statistic. So I had to take a closer look at what I was reading to get the why we needed an array here.

Just having what you wrote afterwards is ok. You don't need to talk about broadcasting or numpy specifics here.

```
# We need to calculate the wald statistics for each combination of x1 and
# x2.
```

```suggestion
```